handbook/engineering/developer-insights/index.md

```diff
@@ -50,17 +50,32 @@ In addition to the [engineering principles and practices](../principles-and-prac
 
 ### Unblocking others is your highest priority
 
-If a teammate is blocked by you on a question, your approval, or a pull request review, your top priority is always to unblock them, either directly or through helping them find someone else who can, even if this takes time away from your own or your team's priorities.
+If a teammate is blocked by you on a question, your approval, or a pull request review, your top priority is always to unblock them, either directly or through helping them find someone else who can, even if this takes time away from your own or your team's priorities. If you're the one who is blocked, be sure to communicate that so that others can prioritize appropriately.
+
+Normally, waiting for a PR review does not mean you are blocked: it's expected that you can start working on something else (e.g. a new PR that depends on the first one). However, in some scenarios it's important to get a review ASAP — e.g., to fix a regression or a broken CI pipeline. In those cases, you should communicate the urgency and expect that your teammates will prioritize unblocking you.
 
 We want teammates to do what is best for the org as a whole. Don't optimize for the goals of your team when it negatively impacts the goals of other teams, our users, and/or the company. Those goals are also your problem and your job.
 
-### Pull request reviews
+### Pull requests
+
+#### For authors
+
+##### Prefer small PRs (<400 lines)
+
+We extend Sourcegraph's company-wide guidance (see [what makes an effective PR](https://docs.sourcegraph.com/dev/background-information/code_reviews#what-makes-an-effective-pull-request-pr)) with a specific guideline that _PRs should contain less than 400 changed lines_ (excluding tests). Note that this is a _guideline_ and not a hard limit; there are situations where it doesn't make sense (e.g. PRs that are mostly mechanical changes).
```

Comment:
Excluding tests will make it a bit harder to self-assess this as GitHub's diff indicator won't do that. Tests need to be reviewed too, maybe omit that rule?

Reply:
We had this on my previous team, and the reasoning was: if tests are included, there will always be an incentive to skimp on tests in order to keep your PRs under the guideline.

But, I also agree with your point. I'm fine with going either way.

Reply:
Makes sense. Just curious, how do we enforce this rule? Is there a linter for this or do we just ask the PR to be broken down when it shows up?

Reply:
@muratsu For now we should just ask for it to be broken up, but a CI check is a good idea!

Reply:
BTW you're right that it's not trivial to assess the number of lines not including tests, but it's also not that hard.  First, it only matters if you are near the limit. Also, GitHub shows the lines changed per file, so you just need to scan through and mentally subtract any tests from the total. In my experience it's not a big deal. You might see a 600-line PR, then spend maybe 10-15 seconds checking how many lines are tests.

Reply:
> We had this on my previous team, and the reasoning was: if tests are included, there will always be an incentive to skimp on tests in order to keep your PRs under the guideline.
> 
> But, I also agree with your point. I'm fine with going either way.

@pdubroy If you had this situation in your previous team, I'd trust your experience.

> Just curious, how do we enforce this rule? Is there a linter for this or do we just ask the PR to be broken down when it shows up?

I think this rule will only work if engineers agree on it and give each other feedback on PRs when it feels out of line (using their judgement to identify exceptions, or when the PR should be even smaller). I _think_ a mechanical check would only make it more likely folks would try to make the metric the goal.

Reply:
Makes sense! 👍

Reply:
Thanks for the feedback! Gonna merge now, and we can revisit this discussion in a few weeks if necessary.